keps/sig-cloud-provider/1959-service-lb-class-field/README.md

```diff
@@ -0,0 +1,213 @@
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [ServiceClass Resource](#serviceclass-resource)
+  - [Generic Annotation](#generic-annotation)
+  - [Provider-Specific Annotations](#provider-specific-annotations)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+When Service Type=LoadBalancer is enabled by a Kubernetes cloud provider, it is a global
+configuration that applies for all Service Type=LoadBalancer resources in a given cluster.
+This becomes problematic if users want to leverage multiple Service Type=LoadBalancer
+implementations in a cluster.
+
+The new [Services APIs](https://github.com/kubernetes-sigs/service-apis) addresses this already
+with the GatewayClass resource. However, until Gateway/GatewayClass APIs become mature, we should
+support similar functionality for Services of Type=LoadBalancer. Introducing a new resource like
+`ServiceClass` is probably not worthwhile given that there are new APIs already in development.
+This KEP proposes a light-weight approach for Service Type=LoadBalancer by introducing a Service
+field `service.spec.loadBalancerClassName`.
```

Comment:
This is super subtle, but convention is that a field called "FoobarName" is the name of a Foobar resource.  Since we're NOT adding a resource here, this should probably just be loadBalancerClass

Reply:
That is subtle but makes a lot of sense, will update.